File: TP2/money_js/src/money.js
var money= (function(){
	function money(v,curr) {
		if(v<0){
			throw new ValueLowerZeroExec();
		}
		this.v=v;
		if(curr.length>3){
			throw new ErrorCurrencyLength();
		}
		this.curr=curr;
	}

	money.prototype.getCurrency=function () {
		return this.curr.toUpperCase();
	}
	money.prototype.getValue=function () {
		return this.v;
	}
	money.prototype.equals=function (otherM) {
		return (otherM.getValue()==this.getValue() && otherM.getCurrency()==this.getCurrency()) ;
	}
	money.prototype.toString=function toString() {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	}


return money;
})();

File: TP1/src/fr/ulille1/fil/odeva/Money.java
package  fr.ulille1.fil.odeva;

public class Money {
	private int value;
	private String currency;
	

	Money(int value, String currency)
	{
		this.value=value;
		this.currency=currency;
	}

	public int getValue()
	{
		return this.value;
	}

	public String getCurrency()
	{
		return this.currency;
	}

    public String toString() {
     	return this.getValue()+" ("+this.getCurrency()+")";
    }
    
    public boolean equals(Object o2){
    	Money m2 = (Money) o2;
		if(this.getValue()==m2.getValue()
				&& this.getCurrency().equals(m2.getCurrency())){
			return true;
		}
    	return false;
    }
    
    public boolean _equals(Object o2){
    	Money m2 = (Money) o2;
		if(this.getValue()==m2.getValue()
				&& this.getCurrency().equals(m2.getCurrency())){
			return true;
		}
    	return false;
    }

}
